faust/agents/replies.py

```diff
@@ -174,6 +174,7 @@ async def _start_fetcher(self, topic_name: str) -> None:
             self._fetchers[topic_name] = None
             # declare the topic
             topic = self._reply_topic(topic_name)
+            self.app.topics.add(topic)
```

Comment:
I'm surprised this isn't called already. I'll need to double-check things.

Reply:
It's also sensitive to ordering. That needs to happen before `topic.maybe_declare()`, or it still fails.

The reply topic is created lazily, it seems. `Conductor::on_client_only_start` runs early, when there are no topics added, so it fails to build the callback map. When `topic.maybe_declare()` does finally ask the producer to create the topic, the conductor doesn't know anything about it, but it still expects there to be a callback installed.

Reply:
I'll need to review how the callbacks are installed. Because if `maybe_declare` fails, then a topic will be erroneously added to the conductor's topic list. It would make more sense to add a topic to the list of conductor topics after successful topic creation like we do with leader topics in https://github.com/faust-streaming/faust/blob/master/faust/assignor/leader_assignor.py#L24-L28. Let me dig into Faust's internals to see if we can install a callback in advance.

Reply:
It seems that it is in fact *not* sensitive to ordering. My bad. I think hit a Kafka re-balancing, and couldn't tell the difference between the latency and it hanging.